### PR TITLE
IDE-881 Switch to Chromium for rendering HTML

### PR DESCRIPTION
**Pull request by GordonSmith**

### CODE REVIEW

**Comment by GordonSmith** [comment]
@dehilsterlexis please review (you may want to look at the second commit only:  https://github.com/hpcc-systems/eclide/pull/467/commits/d55d09b7cebf9ed29cfc086893b23dc2a5bd6bab)

**Comment by dehilsterlexis** [comment]
@GordonSmith rebase?

**Comment by GordonSmith** [comment]
No its dependent on the latest vcpkgs

**Comment by dehilsterlexis** [comment]
@GordonSmith of course! Needs to be added to the VCPKG.

**Comment by dehilsterlexis** [comment]
@GordonSmith @richardkchapman looks good

File: eclide/QueryBuilder.cpp [hunk 1]
```diff
@@ -54,6 +71,22 @@ HMODULE hGrid = 0;
...
+{
+    auto result = CWinAppEx::PumpMessage();
+
+    // If there are other messages on queue then return right away
+    // otherwise CEF has a habit of eating keystrokes not meant for it
+    MSG msg;
+    if (PeekMessage(&msg, NULL, 0, 0, PM_NOREMOVE))
+        return result;
+
+    // Allow Cef to do his thing
```
Comment (GordonSmith) [inline eclide/QueryBuilder.cpp]: Change non PC comment

File: eclide/QueryBuilder.cpp [hunk 2]
```diff
@@ -225,7 +263,26 @@ void CQueryBuilderApp::SaveCustomState()
 {
 }
 
-// CQueryBuilderApp message handlers
-
+bool CQueryBuilderApp::InitializeCef()
+{
+    // Initialize Chrome
```
Comment (GordonSmith) [inline eclide/QueryBuilder.cpp]: Chromium not Chrome

File: eclide/QueryBuilder.cpp [hunk 3]
```diff
@@ -225,7 +263,26 @@ void CQueryBuilderApp::SaveCustomState()
...
+    m_cefApp = new CefMfcCefApp;
+    CefMainArgs mainargs(m_hInstance);
+
+    // See if this is a Chrome process starting. If so, we're done with our (and MFC) initialization.
```
Comment (GordonSmith) [inline eclide/QueryBuilder.cpp]: Chromium

File: eclide/SummaryView.cpp [hunk 4]
```diff
@@ -82,54 +85,55 @@ LRESULT CSummaryView::OnBrowserNew(WORD /*wNotifyCode*/, WORD /*wID*/, HWND /*hW
     return 0;
 }
 
-LRESULT CSummaryView::OnBrowserIE(WORD /*wNotifyCode*/, WORD /*wID*/, HWND /*hWndCtl*/, BOOL& /*bHandled*/)
+LRESULT CSummaryView::OnBrowserNewEclWatch(WORD /*wNotifyCode*/, WORD /*wID*/, HWND /*hWndCtl*/, BOOL& /*bHandled*/)
```
Comment (GordonSmith) [inline eclide/SummaryView.cpp]: I think we should keep the "Open in IE" option for one iteration?

File: en_us/en_us.rc [hunk 5]
```diff
@@ -1220,29 +1248,52 @@ IDB_MAIN_AMT            BITMAP                  "res\\main_amt.bmp"
...
+
 IDI_RADIO_UNCHECKED     ICON                    "res\\radio_unchecked.ico"
+
 IDI_RADIO_CHECKED       ICON                    "res\\radio_checked.ico"
+
 IDI_BUSY                ICON                    "res\\threestate_busy.ico"
+
 IDI_BLANK               ICON                    "res\\threestate_blank.ico"
 
+
```
Comment (GordonSmith) [inline en_us/en_us.rc]: Remove blank spaces

File: wlib/CefClientHandler.cpp [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+{
+}
+
+CefClientHandler::~CefClientHandler()
+{
+}
+
+void CefClientHandler::CreateBrowser(CefWindowInfo const & info, CefBrowserSettings const & settings, CefString const & url)
+{
+    _DBGLOG(LEVEL_INFO, _T("CefClientHandler::CreateBrowser"));
```
Comment (GordonSmith) [inline wlib/CefClientHandler.cpp]: Remove debug logging for entire file...

File: wlib/HtmlChromiumView.h [hunk 7]
```diff
@@ -0,0 +1,320 @@
+// HtmlView.h : interface of the CHtmlChromiumView class
+//
+/////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <file_ver.h>
```
Comment (GordonSmith) [inline wlib/HtmlChromiumView.h]: Is this header needed?

File: wlib/HtmlChromiumView.h [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+#include <Autoupdate.h>
```
Comment (GordonSmith) [inline wlib/HtmlChromiumView.h]: Is this header needed?

File: wlib/HtmlIEView.h [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+        Base64Encode((BYTE *)narrowAuth.c_str(), narrowAuth.length(), buff, &buffLen);
+        extraHeaders += "Authorization: Basic ";
+        extraHeaders += CString(buff, buffLen);
+        //extraHeaders += "\r\n";
+
+        _bstr_t url(m_url.c_str());
+        CComVariant vTargetFrameName = newWindow ? _T("_blank") : _T("_top");
+        CComVariant vFlags;
+        CComVariant vPostData;
+        CComVariant vHeaders(extraHeaders);
```
Comment (GordonSmith) [inline wlib/HtmlIEView.h]: Remove unused code

**Comment by GordonSmith** [comment]
@dehilsterlexis code updated with "optional" chromium support.

File: wlib/HtmlChromiumView.h (hunk 8 above)
Comment (GordonSmith) [reply wlib/HtmlChromiumView.h]: Yes

File: wlib/CefClientHandler.cpp (hunk 6 above)
Comment (GordonSmith) [reply wlib/CefClientHandler.cpp]: Added _DEBUG level

File: wlib/HtmlChromiumView.h (hunk 7 above)
Comment (GordonSmith) [reply wlib/HtmlChromiumView.h]: Nope

**Comment by GordonSmith** [comment]
@richardkchapman ready for review / pull

**Pull request merged**